go/go-sql/go-sql_test.go

```diff
@@ -0,0 +1,87 @@
+package gosql
+
+import (
+	"context"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+var engine, connectionParams = "mysql", "root:root@/gotest"
+
+func TestDisabled(t *testing.T) {
+	mockDB, _, err := sqlmock.New()
+
+	db := DB{DB: mockDB, options: CommenterOptions{}}
+	if err != nil {
+		t.Fatalf("MockSQL failed with unexpected error: %s", err)
+	}
+
+	query := "SELECT 2"
+	if got, want := db.withComment(context.Background(), query), query; got != want {
+		t.Errorf("db.withComment(context.Background(), %q) = %q, want = %q", query, got, want)
+	}
+}
+
+func TestHTTP_Net(t *testing.T) {
+	mockDB, _, err := sqlmock.New()
+
+	db := DB{DB: mockDB, options: CommenterOptions{EnableDBDriver: true, EnableRoute: true, EnableFramework: true}}
+	if err != nil {
+		t.Fatalf("MockSQL failed with unexpected error: %s", err)
+	}
+
+	r, _ := http.NewRequest("GET", "hello/1", nil)
+	ctx := AddHttpRouterTags(r, context.Background())
+
+	got := db.withComment(ctx, "Select 1")
+	want := "Select 1/*driver=database%2Fsql,framework=net%2Fhttp,route=hello%2F1*/"
```

Comment:
This is missing escaping `%` like [Python](https://github.com/google/sqlcommenter/blob/14e94421301f01b2932d83a2857ee8068e3490c1/python/sqlcommenter-python/google/cloud/sqlcommenter/__init__.py#L56-L64) and [PHP](https://github.com/google/sqlcommenter/blob/master/php/sqlcommenter-php/packages/sqlcommenter-laravel/src/Utils.php#L37-L48).

Why not use the same examples their test suites and are shown on the public docs, i.e. testing `congo=t61rcWkgMzE,rojo=00f067aa0ba902b7` becomes `congo%%3Dt61rcWkgMzE%%2Crojo%%3D00f067aa0ba902b7`.